drivers/spi/spi_ll_stm32.c

```diff
@@ -35,6 +35,22 @@
 /* Value to shift out when no application data needs transmitting. */
 #define SPI_STM32_TX_NOP 0x00
 
+#define UNALIGNED_PUT(v, p)					\
+do {								\
+	struct __attribute__((__packed__)) {			\
+		__typeof__(*p) __v;				\
+	} *__p = (__typeof__(__p)) (p);				\
+	__p->__v = (v);						\
+} while (0)
+
+#define UNALIGNED_GET(p)					\
+__extension__ ({						\
+	struct  __attribute__((__packed__)) {			\
+		__typeof__(*(p)) __v;				\
+	} *__p = (__typeof__(__p)) (p);				\
+	__p->__v;						\
+})
+
```

Comment:
Hi,

Why redefining then as they are defined per toolchain and should be available including toolchain.h ?

Reply:
Hi, 
Effectively, including toolchain.h is much better. (In fact, I get them from spi_dw.c file)
A new patch set is coming.